libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -236,14 +240,15 @@ def fmt_err(x: str) -> str:
 
 
 class AzureSearch(VectorStore):
-    """`Azure Cognitive Search` vector store."""
+    """`Azure AI Search` vector store."""
 
     def __init__(
         self,
-        azure_search_endpoint: str,
-        azure_search_key: str,
-        index_name: str,
-        embedding_function: Union[Callable, Embeddings],
+        azure_search_endpoint: str = "",
```

Comment:
are these no longer required?

if so, I'd suggest adding a docstring that explains what parameters (or combinations of parameters) must be passed in, and validating that early in \_\_init\_\_.

IMO it's more clear to have something like
```python
embedding_function: Union[Callable, Embeddings, None] = None,
```
and then raise an error if it's None. ideally if they are required we keep them positional though.

Reply:
azure_search_endpoint, index_name and the embedding_function are all still required. We need to make the azure_search_key optional, since with managed identity the user passes in a token instead. 

We could leave azure_search_endpoint as positional, but my main concern was that if I left index_name  and embedding_function as positional then the order of the positional arguments would change because index_name would need to come before azure_search_key, which could break things for current users. Making things optional means that we can make this change backwards compatible. Not sure how else to do this without breaking things for users. What do you think @ccurme?

Reply:
I think for this I was trying to weigh the what makes more sense, keeping required arguments positional or prioritising maintaining the order of the arguments for the sake of backwards compatibility. Open to thoughts on what is preferred, or other ideas in case I'm missing something.

Reply:
Yeah that makes sense. I'd prioritize backwards compatibility here. My first thought is:

- `azure_search_endpoint` stays positional (why not)
- `azure_search_key`, `index_name`, `embedding_function` default to None similar to `azure_credential`
- We add a short function to validate parameters and call it early in `__init__`
  - tests that either `azure_search_key` or `azure_credential` are provided
  - tests that `index_name` and `embedding_function` are provided
- add a docstring explaining required arguments

Let me know what you think.

Reply:
This makes sense to me!!! I'll start working on these changes!